shell/platform/darwin/macos/framework/Source/FlutterEngine.mm

```diff
@@ -603,6 +654,48 @@ - (FlutterPlatformViewController*)platformViewController {
 
 #pragma mark - Private methods
 
+- (FlutterEngineViewRecord*)viewRecordForId:(uint64_t)viewId {
+  if (viewId == kFlutterDefaultViewId) {
+    return _viewRecord;
+  }
+  return nil;
+}
+
+- (void)updateSemantics:(const FlutterSemanticsUpdate*)update forViewId:(uint64_t)viewId {
+  FlutterEngineViewRecord* record = [self viewRecordForId:viewId];
+  if (record == nil) {
+    return;
+  }
```

Comment:
Should the rest of this method be moved to the `FlutterEngineViewRecord` or `FlutterViewController`? It seems reasonable for the engine to receive semantics updates and dispatch them to the view, but actually applying the semantics update to a view seems like it should be outside the engine's scope.

Reply:
The entire `updateSemantics` has been moved to `FVC`.